Use `target` instead of `machine` for mir interpreter integer handling.

oli-obk opened the pull request:
The naming of `machine` only makes sense from a mir interpreter internals perspective, but outside users talk about the `target` platform. As per https://github.com/rust-lang/rust/pull/108029#issuecomment-1429791015

r? @RalfJung

RalfJung commented on compiler/rustc_const_eval/src/const_eval/target.rs:
```diff
@@ -0,0 +1,656 @@
+use rustc_hir::def::DefKind;
```
What happened here, why is there such a new file?

oli-obk replied:
huh... that was in my edits during branch switching and probably got saved by vscode after the branch switch removed it

oli-obk committed:
Use target instead of machine for mir interpreter integer handling.

The naming of `machine` only makes sense from a mir interpreter internals perspective, but outside users talk about the `target` platform

RalfJung commented:
So this renames machine_usize → target_usize but leaves other uses of the term "machine" unchanged? LGTM, r=me with CI happy.

oli-obk commented:
> leaves other uses of the term "machine" unchanged

yea. I looked at the other uses, like the trait and modules, but it's not really about the target. The `Machine` trait impls for example are different mir interpreters and, while internally handling target specific things, that's not their purpose. There are no more `fn .*machine.*` except those that refer to LLVM or GCC specific backend logic.

RalfJung commented:
> The Machine trait impls 

Those are about the "Abstract Machine" and I think the name still makes sense.

bors commented:
:umbrella: The latest upstream changes (presumably #108052) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

RalfJung commented:
@bors r+

bors commented:
:pushpin: Commit 38b7cdf393737ee6702a2b4cc01275c6bffbd3e2 has been approved by `RalfJung`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=RalfJung 38b7cdf393737ee6702a2b4cc01275c6bffbd3e2 -->
<!-- homu: {"type":"Approved","sha":"38b7cdf393737ee6702a2b4cc01275c6bffbd3e2","approver":"RalfJung","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
